tests/basic/storage_pool_test.go

```diff
@@ -2053,4 +2056,597 @@ var _ = Describe("{VolUpdateAddDrive}", func() {
+		var selectedNode node.Node
+		isMultiPoolNode := false
+		for _, selNode := range stNodes {
+			log.InfoD("Validating Node %s for multipool configuraitons", selNode.Name)
+			if len(selNode.StoragePools) > 1 {
+				isMultiPoolNode = true
+				selectedNode = selNode
+				break
+			}
+		}
+
+		dash.VerifyFatal(isMultiPoolNode, true, "Failed as Multipool configuration doesnot exists!")
+		// Selecting Storage pool based on Pools present on the Node
+		selectedPool, err := GetPoolWithIOsInGivenNode(selectedNode)
+		log.FailOnError(err, "error while selecting the pool [%s]", selectedPool)
+
+		stepLog = fmt.Sprintf("Expanding pool on node %s and pool UUID: %s using auto", selectedNode.Name, selectedPool.Uuid)
+		Step(stepLog, func() {
+			poolToBeResized, err := GetStoragePoolByUUID(selectedPool.Uuid)
+			log.FailOnError(err, "Failed to get pool using UUID [%s]", selectedPool.Uuid)
+			expectedSize := poolToBeResized.TotalSize * 2 / units.GiB
+
+			isjournal, err := isJournalEnabled()
+			log.FailOnError(err, "Failed to check if Journal enabled")
+
+			log.InfoD("Current Size of the pool %s is %d", selectedPool.Uuid, poolToBeResized.TotalSize/units.GiB)
+			err = Inst().V.ExpandPool(selectedPool.Uuid, api.SdkStoragePool_RESIZE_TYPE_AUTO, expectedSize)
+			dash.VerifyFatal(err, nil, "Pool expansion init successful?")
+
+			resizeErr := waitForPoolToBeResized(expectedSize, selectedPool.Uuid, isjournal)
+			dash.VerifyFatal(resizeErr, nil, fmt.Sprintf("Verify pool %s on node %s expansion using auto", selectedPool.Uuid, selectedNode.Name))
+		})
+
+		opts := make(map[string]bool)
+		opts[scheduler.OptionsWaitForResourceLeakCleanup] = true
+		ValidateAndDestroy(contexts, opts)
+
+		stepLog = "destroy all the applications created before test runs"
+		Step(stepLog, func() {
+			log.InfoD(stepLog)
+			opts := make(map[string]bool)
+			opts[scheduler.OptionsWaitForResourceLeakCleanup] = true
+			for _, ctx := range contexts {
+				TearDownContext(ctx, opts)
+			}
+		})
+	})
+	JustAfterEach(func() {
+		defer EndTorpedoTest()
+		AfterEachTest(contexts, testrailID, runID)
+	})
+})
+
+var _ = Describe("{CreateSnapshotsPoolResize}", func() {
+	/*
+		Steps : Try pool resize when ot of snapshots are created on the volume
+
+	*/
+	var testrailID = 50652
+	// Testrail Description : Try pool resize when lot of snapshots are created on the volume
+
+	var runID int
+
+	JustBeforeEach(func() {
+		StartTorpedoTest("CreateSnapshotsPoolResize", "Validate storage pool expansion when lots of snapshots present on the system", nil, testrailID)
+		runID = testrailuttils.AddRunsToMilestone(testrailID)
+	})
+
+	var contexts []*scheduler.Context
+	var totalSnapshotsPerVol int = 60
+
+	snapshotList := make(map[string][]string)
+	var selectedNode node.Node
+	var pickNode string
+
+	// Try pool resize when ot of snapshots are created on the volume
+	stepLog := "should get the existing storage node and expand the pool by resize-disk"
+
+	It(stepLog, func() {
+		log.InfoD(stepLog)
```

Comment:
Same as above, there is a lot of similar place you have done this, fix it for every step please